java/core/src/main/java/sleeper/core/key/KeySerDe.java

```diff
@@ -59,7 +59,7 @@ public KeySerDe(List<PrimitiveType> rowKeyTypes) {
     public byte[] serialise(Key key) throws IOException {
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
         DataOutputStream dos = new DataOutputStream(baos);
-        int numKeys = key.size();
+        int numKeys = key != null ? key.size() : 0;
         int numKeysToSerialise = Math.min(numKeys, numRowKeysInSchema);
```

Comment:
These changes mean that a key consisting of an empty list and a null key get serialised to the same thing, which doesn't seem right. However we're likely to remove KeySerDe when https://github.com/gchq/sleeper/issues/1378 is done so we can ignore this for now.